kernel/kernel-api/src/main/java/io/delta/kernel/internal/checkpoints/CheckpointInstance.java

```diff
@@ -117,14 +168,26 @@ public boolean equals(Object o) {
         if (o == null || getClass() != o.getClass()) {
             return false;
         }
+        // For V2 checkpoints, compare the filepath.
         CheckpointInstance checkpointInstance = (CheckpointInstance) o;
+        if (checkpointInstance.format == CheckpointFormat.V2 && format == CheckpointFormat.V2 &&
+                !filePath.equals(checkpointInstance.filePath)) {
+            return false;
+        }
         return version == checkpointInstance.version &&
-                Objects.equals(numParts, checkpointInstance.numParts);
+                Objects.equals(numParts, checkpointInstance.numParts) &&
+                format == checkpointInstance.format;
```

Comment:
can we use the `compareTo() == 0` here?

Reply:
Fixed (for non-optional fields, optional still requires `Objects.equals`)